source/Dgraph/Channels/CloudChannel.cs

```diff
@@ -35,11 +44,39 @@ public static GrpcChannel Create(string address, string apiKey)
 
             // SslCredentials is used here because this channel is using TLS.
             // CallCredentials can't be used with ChannelCredentials.Insecure on non-TLS channels.
-            var channel = GrpcChannel.ForAddress(address, new GrpcChannelOptions
+            return GrpcChannel.ForAddress(address, new GrpcChannelOptions
             {
                 Credentials = ChannelCredentials.Create(new SslCredentials(), credentials)
             });
-            return channel;
+        }
+
+        internal static string ProcessAddress(string address)
+        {
+            if (!address.Contains("cloud.dgraph.io"))
+            {
+                throw new ArgumentException("The provided address is not valid for Dgraph Cloud. Check your Cloud gRPC Endpoint.");
+            }
+            if (address.StartsWith("http://") || address.StartsWith("https://"))
+            {
+                address = new Uri(address).Host;
+            }
+
+            var match = Regex.Match(address, @"^(?<name>[\w-]+)\.(?<region>[\w-]+)\.aws\.cloud\.dgraph\.io");
```

Comment:
let's not include this in this PR. I have doubts that aws may not always be there in the address. We will merge it in the separate PR

Reply:
Ok I reverted back to my original implementation for now. It is similar to how dgo handles parsing the address.